Bump Go to 1.13.8

remeh opened the pull request:
### What does this PR do?

Use latest version available of Go.

### Motivation

We want to benefit from latest improvements.

### Additional Notes

We've noticed that the new builders are not generating exactly the same
`tracer-ebpf.o` output. That may be due to a slight package change in clang
(8.0.1-7 -> 8.0.1-8 even if a bit unlikely as it is a package change).
We've validated that the tests are still passing properly with the new binary and
we're then updating it.

Link to public builders change: DataDog/datadog-agent-buildimages#30

 - [x] Must not be merged for `7.18.0`, we want to ship this with `7.19.0`.
 - [x] Still needs to update the builder tags in `gitlab.yaml`
 - [x] CircleCI image has to be pushed

remeh committed:
[go] bump Go to 1.13.6

remeh committed:
circleci: bump go to 1.13.6

remeh committed:
general: release note entry for go1.13

remeh committed:
dogstatsd: don't ship the block profiler enabling.

remeh committed:
[build] bump to 1.13.8

remeh committed:
[tasks] fix integration tests after upgrade to Go 1.13

Since Go 1.13, the -exec flag of go test could add some parameters such as -test.timeout to the call,
we don't want them because while calling invoke below, invoke thinks that the parameters are for it to interpret.
We're now calling an intermediate script which only pass the binary name to the invoke task.

olivielpeau reviewed (dismissed):
Thanks, looks good to merge once CI is green, the gitlab runner tags are updated and a full pipeline passes 👍

hush-hush reviewed (changes requested):
Should we not also update the gitlab builders ?

remeh committed:
[gitlab] update builders.

remeh committed:
Merge remote-tracking branch 'origin/master' into remeh/go-1.13.6

remeh committed:
[gitlab] provide Go version to system-probe build.

remeh committed:
[gitlab] use latest builders shipping the fix for libbcc jobs.

remeh committed:
[ebpf] update `tracer-ebpf.go` due to the changes in the builders.

We've noticed that the new builders are not generating exactly the same
`tracer-ebpf.o` output. That may be due to a slight package change in clang
(8.0.1-7 -> 8.0.1-8 even if a bit unlikely as it is a package change).
We've validated that the tests are still passing properly with the new binary and
we're then updating it.

olivielpeau commented on releasenotes/notes/go-1-13-8-d98857855401db29.yaml:
```diff
@@ -0,0 +1,4 @@
+---
+other:
+  - |
+    The Agent, Logs Agent and the system-probe are now compiled with Go ``1.13.8``
```
Also APM and process agents no?

(in that case, I guess writing `all the Agent binaries` makes sense)

remeh replied:
Got a bit too much excited by seeing all those ✅
Let me send another quick PR for that...

olivielpeau reviewed (approved):
LGTM once the gitlab pipeline is ✅

Just one question on the release note.

What's the QA plan for this? (we can follow up on the card)
